libpod/pod.go

```diff
@@ -158,6 +221,30 @@ func (p *Pod) AllContainers() ([]*Container, error) {
 	return p.runtime.state.PodContainers(p)
 }
 
+// HasPauseContainer returns whether the pod will create a pause container
+func (p *Pod) HasPauseContainer() bool {
+	return p.config.PauseContainer.HasPauseContainer
+}
+
+// SharesNamespaces checks if the pod has any kernel namespaces set as shared. A pause container will not be
+// created if no kernel namespaces are shared.
+func (p *Pod) SharesNamespaces() bool {
+	return p.SharesPID() || p.SharesIPC() || p.SharesNet() || p.SharesMNT() || p.SharesUser() || p.SharesUTS()
+}
+
+// PauseContainerID returns a the pause container ID for a pod.
+// If the container returned is "", the pod has no pause container.
+func (p *Pod) PauseContainerID() (string, error) {
+	p.lock.Lock()
+	defer p.lock.Unlock()
+
```

Comment:
Need to force a sync here with updatePod()

Reply:
Basically - replace the p.valid check with updatePod(), and remove updatePod from pauseContainerID.